rir/src/compiler/opt/constantfold.cpp

```diff
@@ -106,6 +121,10 @@ void Constantfold::apply(RirCompiler& cmp, Closure* function,
                     branchRemoval.emplace(bb, a == b);
                 });
             }
+
+            if (auto isObj = IsObject::Cast(condition))
```

Comment:
this is correct, because asTest and asLogical are both effectful. asLogical prints a warning if it is not scalar and asTest throws an error if it's NA. That's just how R works. So e.g. it could be that we remove the branch, but still need to show a warning that the branch condition was larger than 1.